### DISCUSSION TITLE
Insert multiple rows of data at once

### DISCUSSION THREAD

**Discussion by edikurniawan-dev** [pocketbase/pocketbase#2907] [Q&A] [answered]
I have around 400 rows of data like below which I get hourly
```
{
  "collectTime": 1685552400000,
  "code": "337",
  "dataItemMap": {
    "radiation_intensity": null,
    "installed_capacity": 5.82,
  }
},
  .......
{
  "collectTime": 1685638800000,
  "code": "337",
  "dataItemMap": {
    "radiation_intensity": null,
    "installed_capacity": 5.82,
  }
},
```
Is it possible to insert multiple rows of data at once in pocketbase?

**Comment by ganigeorgiev** [comment, marked as answer]
No, currently the web APIs don't support bulk operations. It is in my todo and will be implemented sometime in the near future as part of #48.

As a workaround, you can [register a custom route](https://pocketbase.io/docs/router/#register-new-route) and perform the db operations on your own (eg. in a [transaction](https://pocketbase.io/docs/database/#transaction)). More detailed documentation how to use PocketBase as framework (including js app hooks) will be available with the upcoming v0.17.0 release soon.

**Reply by edikurniawan-dev** [reply]
Oh i see,
How many data row limits are there on pocketbase?

**Reply by ganigeorgiev** [reply]
It is difficult to say exactly. For more details you can check [Limits in SQLite](https://www.sqlite.org/limits.html). Note that if your application is write heavy PocketBase may not perform well depending on your use case and how do you plan interacting with your data. You can also check the recent benchmarks in https://github.com/pocketbase/benchmarks. For example on Hetzner CAX11 (2vCPU, 4GB RAM) the web APIs performance when inserting 5000 records with 500 concurrent requests is:

```
┌─ Best:      5.788378ms
├─ Worst:     5.662470531s
├─ Completed: 6.137892547s
└─ Errors:    0
```
_(with lower more reasonable concurrency most insert queries perform close to the "Best" section)_

Which I guess should be fine for small and midsize applications, but again it depends on your use case and how do you plan interacting with your data, because at the moment we don't really have much filters for timeseries type of data (_or at leat that what it looks like from your example_).
